src/services/ipfs-service.js

```diff
@@ -31,40 +31,86 @@ class IpfsService {
     this.mapCache = new MapCache()
   }
 
-  async submitFile(jsonData) {
+  /**
+   * Convert an object to a JSON blob and submit it to IPFS.
+   *
+   * obj {string} - object to save as file
+   */
+  async saveObjAsFile(obj) {
+    let file
+    if (typeof Blob === 'undefined') {
+      file = new Buffer(JSON.stringify(obj))
+    } else {
+      file = new Blob([JSON.stringify(obj)])
+    }
+
+    const ipfsHash = await this.saveFile(file)
+    // Cache the object
+    this.mapCache.set(ipfsHash, obj)
```

Comment:
Couple questions on the caching:
 - Are we actually getting any benefit from it ? Since we are retrieving the IPFS blobs via http, doesn't the browser cache this data ?
 - If caching does help, should we not only cache JSON data but also image data ?

Reply:
Good question. I just kept it around because it was how it was being done previously.

The image data is not loaded through `origin-js` like the JSON data is (i.e. there is no load function for images). The URLs are just used in the dapp so the browser just handles caching like standard image URLs.

Reply:
Ah yes, the images don't get loaded via origin-js so no point in caching those.
Let's leave the cache as is for now in this PR. And we can investigate as a separate effort whether cache helps or not later.